### PR TITLE
Filecoin support

### PR DESCRIPTION
**Pull request by makoto**
## Issue number

https://github.com/ensdomains/address-encoder/issues/85

## Reference to the specification

- [filecoin project spec page](https://spec.filecoin.io/#section-glossary.address)
- [go implementation](https://github.com/filecoin-project/go-address/blob/master/address.go)

## List of features added/changed

-  Include `@glif/filecoin-address` and define module
- Add filAddrEncoder/filAddrDecoder
- Add test

## How Has This Been Tested?

Unit test 

## Anything 

The file size increased from  760188  to 861221 as it included external lib

## Screenshots (if appropriate):

Screenshot 2020-10-27 at 15 37 49

## Checklist:

- [ x] My code follows the code style of this project.
- [ x] My code implements all the required features.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -329,6 +336,17 @@ function ksmAddrDecoder(data: string): Buffer {
   return new Buffer(ss58Decode(data));
 }
 
+function filAddrEncoder(data: Buffer): string {
+  const address = filNewAddress('1', data)
```
Comment (Arachnid) [inline src/index.ts:340]: Pretty sure we shouldn't be hardcoding version 1 here. From looking at the library, it looks like the first byte in the binary format is the version byte, and we should probably be preserving that instead.

File: README.md [hunk 2]
```diff
@@ -45,6 +45,7 @@ This library currently supports the following cryptocurrencies and address forma
  - EOS
  - ETC (checksummed-hex)
  - ETH (checksummed-hex)
+ - FIL ([bls](https://spec.filecoin.io/#section-glossary.address))
```
Comment (Arachnid) [inline README.md:48]: BLS is the signature scheme, not the address encoding; the address encoding is base32 + blake2b checksum

**Review by Arachnid — CHANGES_REQUESTED** [review]
100k is a *lot* for a single cryptocurrency, and looking at the imported library it looks like the address format is fairly straightforward and only relies on blake2b and base32 - both of which I think we already import for other coins. Would it be possible to import the relevant code (with license/citation) into the address-encoding library instead?

**Comment by makoto** [comment]
@Arachnid when I ported the library and reused `blake2b` and `base32`, the file size went back to `762186`. https://travis-ci.com/github/ensdomains/address-encoder/jobs/428828727
However , filecoin protocol 0 (ID) requires `leb128` lib and with that the file size went back to `856420`. Not sure if porting also `leb128` will reduce file sizes as there aren't overwrapping libraries we can reuse https://gitlab.com/mjbecze/leb128/-/blob/master/package.json .

**Comment by Arachnid** [comment]
It looks like leb128 is only as large as it is because it imports bn.js. Surely that must already be in our transitive dependencies somewhere - possibly a different version?

**Comment by makoto** [comment]
Ok, ported `leb128`. The filesize is now shrunk a bit `760188 -> 732656` for some reason.

**Review by Arachnid — APPROVED** [review]

**Pull request merged**